internal/state/artifact.go

```diff
@@ -0,0 +1,111 @@
+package state
+
+import (
+	"reflect"
+)
+
+// ArtifactReader defines an interface for reading artifact data
+type ArtifactReader interface {
+	GetRepository() string
+	GetTags() []string
+	GetDigest() string
+	GetType() string
+	IsDeleted() bool
+	HasChanged(newArtifact ArtifactReader) bool
+	SetRepository(repository string)
+	SetName(name string)
+	GetName() string
+}
+
+// Artifact represents an artifact object in the registry
+type Artifact struct {
+	Repository string   `json:"repository,omitempty"`
+	Tags       []string `json:"tag,omitempty"`
+	Labels     []string `json:"labels"`
+	Type       string   `json:"type,omitempty"`
+	Digest     string   `json:"digest,omitempty"`
+	Deleted    bool     `json:"deleted"`
+	Name       string   `json:"name,omitempty"`
+}
+
+// NewArtifact creates a new Artifact object
```

Comment:
Remove ai gen comments.

Reply:
Thanks for review @bupd , have made some changes based on your review.